demos/common/cpp/models/src/classification_model.cpp

```diff
@@ -64,84 +66,104 @@ std::vector<std::string> ClassificationModel::loadLabels(const std::string& labe
     return labels;
 }
 
-void ClassificationModel::prepareInputsOutputs(InferenceEngine::CNNNetwork& cnnNetwork) {
+void ClassificationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model) {
     // --------------------------- Configure input & output -------------------------------------------------
-    // --------------------------- Prepare input blobs ------------------------------------------------------
-    InferenceEngine::ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
-    if (inputShapes.size() != 1)
-        throw std::runtime_error("Demo supports topologies only with 1 input");
-    inputsNames.push_back(inputShapes.begin()->first);
-    InferenceEngine::SizeVector& inSizeVector = inputShapes.begin()->second;
-    if (inSizeVector.size() != 4 || inSizeVector[1] != 3)
-        throw std::runtime_error("3-channel 4-dimensional model's input is expected");
-    if (inSizeVector[2] != inSizeVector[3])
+    // --------------------------- Prepare input  ------------------------------------------------------
+    const ov::OutputVector& inputsInfo = model->inputs();
+    if (inputsInfo.size() != 1) {
+        throw std::logic_error("Classification model wrapper supports topologies only with 1 input");
+    }
+    inputsNames.push_back(model->input().get_any_name());
+
+    const ov::Shape& inputShape = model->input().get_shape();
+    ov::Layout inputLayout = ov::layout::get_layout(model->input());
+    if (inputLayout.empty()) {
+        inputLayout = { "NCHW" };
+    }
+
+    if (inputShape.size() != 4 || inputShape[1] != 3) {
```

Comment:
You should access channel number using layout indexes. Same for width and height. Otherwise there's no much meaning in having layout at all

Reply:
Yeah, I am going to update it, thanks.